recipes/dev/generate_v2.py

```diff
@@ -0,0 +1,193 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import itertools
+import sys
+import time
+from typing import Any, Dict, List
+
+import torch
+from omegaconf import DictConfig
+
+from torchtune import config, training, utils
+from torchtune.data import (
+    left_pad_sequence,
+    load_image,
+    Message,
+    padded_collate_tiled_images_and_mask,
+)
+
+from torchtune.generation import sample
+
+from torchtune.modules.transforms import Transform
+
+
+class SingleTurnYAMLToMessages(Transform):
```

Comment:
Should this end up somewhere like `data/_utils.py`?

Reply:
Yep, eventually will move it there.